### PR TITLE
Fixing half-precision alpha for gemm tests.

### PR DESCRIPTION
**Pull request by daineAMD**
Previously wasn't handling this correctly, the simple cast doesn't work as expected. This matches gemm_ex tests for now. Also it doesn't look like we have tests for a non-zero imaginary part for cgemm/zgemm, I'll make a ticket for this.

### CODE REVIEW

File: clients/include/testing_gemm_strided_batched.hpp [hunk 1]
```diff
@@ -44,8 +44,22 @@ hipblasStatus_t testing_gemm_strided_batched(const Arguments& argus)
...
+    if(is_same<T, hipblasHalf>::value)
+    {
+        alpha = float_to_half(alpha_float);
+        beta  = float_to_half(beta_float);
+    }
+    else
+    {
+        alpha = static_cast<T>(alpha_float);
+        beta  = static_cast<T>(beta_float);
+    }
```
Comment (TorreZuk) [inline clients/include/testing_gemm_strided_batched.hpp:62]: Not for this release but expect this should move to use helpers the same as rocblas.  If you can get_alpha<T> and get_beta<T> you can handle this with specializations along with NaN support.

**Review by TorreZuk — APPROVED** [review]
LGTM if the tests pass

**Comment by daineAMD** [comment]
Passed everyone save for the rocSOLVER centos CI issue - merging.
